### PR TITLE
Adding Turbo Features & Specs..

### PR DESCRIPTION
**Pull request by hamzajg**

### CODE REVIEW

**Comment by VaughnVernon** [comment]
@hamzajg I'll need @tjaskula to do all .NET reviews.

File: src/Vlingo.Xoom.Turbo.Tests/Scooter/Model/Object/Employee.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+using Vlingo.Xoom.Lattice.Model;
+
+namespace Vlingo.Xoom.Turbo.Tests.Scooter.Model.Object
+{
+  public interface Employee
```
Comment (tjaskula) [inline src/Vlingo.Xoom.Turbo.Tests/Scooter/Model/Object/Employee.cs]: Should respect .NET convention with `I` in the name prefix

Comment (tjaskula) [reply src/Vlingo.Xoom.Turbo.Tests/Scooter/Model/Object/Employee.cs]: Spacing is off in all the files. It's not blocking but if you can use tab (4 spaces) like in the `BootTest` file

File: src/Vlingo.Xoom.Turbo.Tests/Scooter/Model/Object/Employee.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  {
+    void Assign(string number);
+    void Adjust(int salary);
+    void Hire(string number, int salary);
+
+    public class EmployeeHired : DomainEvent
```
Comment (tjaskula) [inline src/Vlingo.Xoom.Turbo.Tests/Scooter/Model/Object/Employee.cs]: Could you move out classes outside of interface declaration ?

File: src/Vlingo.Xoom.Turbo.Tests/Scooter/Model/Object/EmployeeEntity.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    public EmployeeEntity(long id)
+    {
+      _employee = new EmployeeState(id, "", 0);
+    }
+
+    public void Assign(string number)
+    {
+      throw new System.NotImplementedException();
```
Comment (tjaskula) [inline src/Vlingo.Xoom.Turbo.Tests/Scooter/Model/Object/EmployeeEntity.cs]: Is this normal it is not implemented here ?

Comment (tjaskula) [reply src/Vlingo.Xoom.Turbo.Tests/Scooter/Model/Object/EmployeeEntity.cs]: And in other methods ?

File: src/Vlingo.Xoom.Turbo.Tests/Scooter/Model/Object/EmployeeState.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
+using System.Collections.Generic;
```
Comment (tjaskula) [inline src/Vlingo.Xoom.Turbo.Tests/Scooter/Model/Object/EmployeeState.cs]: Think in all the files adding headers

```
// Copyright © 2012-2021 VLINGO LABS. All rights reserved.
 //
 // This Source Code Form is subject to the terms of the
 // Mozilla Public License, v. 2.0. If a copy of the MPL
 // was not distributed with this file, You can obtain
 // one at https://mozilla.org/MPL/2.0/.
 ```

File: src/Vlingo.Xoom.Turbo.Tests/Scooter/Model/Object/EmployeeState.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+    public EmployeeState(long id, string number, int salary) : base(id)
+    {
+      Number = number;
+      Salary = salary;
+    }
+
+    public bool Equals(EmployeeState x, EmployeeState y)
+    {
+      throw new System.NotImplementedException();
```
Comment (tjaskula) [inline src/Vlingo.Xoom.Turbo.Tests/Scooter/Model/Object/EmployeeState.cs]: Seems like it should be implemented and other equality methods with gethashcode

File: src/Vlingo.Xoom.Turbo.Tests/Storage/MockDispatcher.cs [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+using Vlingo.Xoom.Symbio.Store.Dispatch;
+
+namespace Vlingo.Xoom.Turbo.Tests.Storage
+{
+  public class MockDispatcher<E, RS> : IDispatcher
+  {
+    public void ControlWith(IDispatcherControl control)
+    {
+      throw new System.NotImplementedException();
```
Comment (tjaskula) [inline src/Vlingo.Xoom.Turbo.Tests/Storage/MockDispatcher.cs]: normal ?

File: src/Vlingo.Xoom.Turbo.Tests/Storage/StoreActorBuilderTest.cs [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+    public StoreActorBuilderTest()
+    {
+      _world = World.StartWithDefaults("store-actor-build-tests");
+    }
+
+    [Fact]
+    public void TestThatMySqlJournalActorIsBuilt()
+    {
+      // var journal = StoreActorBuilder.From(_world.Stage, new Model(ModelType.Command.ToString()),
```
Comment (tjaskula) [inline src/Vlingo.Xoom.Turbo.Tests/Storage/StoreActorBuilderTest.cs:27]: Skip the test, don't comment the code, unless it doesn't compile

**Review by tjaskula — CHANGES_REQUESTED** [review]
Some questions and minor requested changes

File: src/Vlingo.Xoom.Turbo.Tests/Storage/StoreActorBuilderTest.cs (hunk 7 above)
Comment (hamzajg) [reply src/Vlingo.Xoom.Turbo.Tests/Storage/StoreActorBuilderTest.cs:27]: yes,  it doesn't compile and the production code is not implemented yet..

**Comment by VaughnVernon** [comment]
@tjaskula I don't know what you prefer, but when using the `I` interface name prefix, there's no need to use a secondary name for the implementation class (which you know). This is probably a good time to establish your convention for entity/aggregate names:

Java:

Interface: `Investor` and  Entity: `InvestorEntity`

.NET: 
1. `IInvestor` and `InvestorEntity` or
2. `IInvestor` and `Investor`
3. Then there is the Designer Aggregate name:
`Investor` and we prepend the `I`
or make the Aggregate name purely. NET with model lockin?
4. It could be that we check for leading `I` and if not given we prepend, and if already there we use it as is.

**Comment by tjaskula** [comment]
@VaughnVernon I think `I` can be appended automatically like a mix between 1 and 4 ?

**Comment by VaughnVernon** [comment]
> @VaughnVernon I think `I` can be appended automatically like a mix between 1 and 4 ?

Ok, to confirm: You prefer 1 over 2 and a blend of 1 and 4?

**Comment by tjaskula** [comment]
I'm not sure whether 1 or 2 is better, but definitely both can be used. I think we can align this with JVM. However definitely we need 4

3 seems ok but I didn't understand `or make the Aggregate name purely. NET with model lockin?`

**Comment by VaughnVernon** [comment]
Option 3 means that the user makes a choice to use `Investor` or `IInvestor` and we use that exactly, making no automatic changes based on selected platform, Java or .NET. It's basically what we have now although we support only Java.

IMO the Designer model should be abstract and the tooling should adjust for the target platform.

File: src/Vlingo.Xoom.Turbo.Tests/Scooter/Model/Object/Employee.cs (hunk 1 above)
Comment (hamzajg) [reply src/Vlingo.Xoom.Turbo.Tests/Scooter/Model/Object/Employee.cs]: I used the same indentation in the JVM side (2 spaces), I will re-configure the IDE to use (4 spaces) Tab..

**Comment by tjaskula** [comment]
@hamzajg just ping me when you think I can review it and merge.

**Pull request merged**